components/filter/filter.js

```diff
@@ -379,12 +466,32 @@ class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
 	_isDimensionEmpty(dimension) {
 		switch (dimension.type) {
 			case 'd2l-filter-dimension-set':
-				return dimension.values.length === 0;
+				return dimension.values.length === 0 && !dimension.searchValue;
 		}
 
 		return false;
 	}
 
+	_searchDimension(dimension) {
+		switch (dimension.type) {
+			case 'd2l-filter-dimension-set':
+				dimension.values.forEach(value => {
+					if (dimension.searchValue === '') {
+						value.hidden = false;
+					} else {
+						if (value.text.toLowerCase().indexOf(dimension.searchValue.toLowerCase()) > -1) {
+							value.hidden = false;
+						} else {
+							value.hidden = true;
+						}
+					}
```

Comment:
Nit: could likely be simplified to:

```suggestion
					value.hidden = !(value.text.toLowerCase().indexOf(dimension.searchValue.toLowerCase()) > -1);
```